crates/turborepo-ui/src/tui/app.rs

```diff
@@ -7,102 +8,265 @@ use std::{
 use ratatui::{
     backend::{Backend, CrosstermBackend},
     layout::{Constraint, Layout},
+    widgets::TableState,
     Frame, Terminal,
 };
 use tracing::debug;
 
 const PANE_SIZE_RATIO: f32 = 3.0 / 4.0;
 const FRAMERATE: Duration = Duration::from_millis(3);
 
-use super::{input, AppReceiver, Error, Event, InputOptions, TaskTable, TerminalPane};
+use super::{
+    event::TaskResult, input, AppReceiver, Error, Event, InputOptions, TaskTable, TerminalPane,
+};
+use crate::tui::{
+    task::{Task, TasksByStatus},
+    term_output::TerminalOutput,
+};
 
-pub struct App<I> {
-    table: TaskTable,
-    pane: TerminalPane<I>,
-    done: bool,
+pub enum LayoutSections {
+    Pane,
+    TaskList,
+}
+
+pub struct App<W> {
+    // TODO: Could I reasonably get to only one representation of the tasks?
+    tasks: BTreeMap<String, TerminalOutput<W>>,
```

Comment:
There is the possibility here, but that should be punted on. 

We would need a way to track order of the tasks which is easiest to due with just having a list structure. Due to the frequent insert/removals which result in lots of shifting of elements we want to keep whatever is in the list on the small side to keep the size of `memcpy` calls small. Putting the whole `TermialOutput` in the list representation would be a perf hit.

In the future we might find the proper data structure that allows us to do this cheap *or* we end up assigning each task an integer id and pass those around.